tools/db_bench_tool.cc

```diff
@@ -5671,17 +5681,39 @@ class Benchmark {
       }
       Status s;
       pinnable_val.Reset();
+      for (size_t i = 0; i < pinnable_vals.size(); ++i) {
+        pinnable_vals[i].Reset();
+      }
+      ColumnFamilyHandle* cfh;
       if (FLAGS_num_column_families > 1) {
-        s = db_with_cfh->db->Get(options, db_with_cfh->GetCfh(key_rand), key,
-                                 &pinnable_val, ts_ptr);
+        cfh = db_with_cfh->GetCfh(key_rand);
       } else {
-        s = db_with_cfh->db->Get(options,
-                                 db_with_cfh->db->DefaultColumnFamily(), key,
-                                 &pinnable_val, ts_ptr);
+        cfh = db_with_cfh->db->DefaultColumnFamily();
+      }
+      if (read_operands_) {
+        GetMergeOperandsOptions get_merge_operands_options;
+        get_merge_operands_options.expected_max_number_of_operands =
+            static_cast<int>(pinnable_vals.size());
+        int number_of_operands;
+        s = db_with_cfh->db->GetMergeOperands(
+            options, cfh, key, pinnable_vals.data(),
+            &get_merge_operands_options, &number_of_operands);
+        if (s.IsIncomplete()) {
+          // OK if it happens a few times.
+          pinnable_vals.resize(number_of_operands);
+          s = Status::OK();
+          continue;
```

Comment:
Why the continue?  Do you not want to count this as a found key or as an operation?

Reply:
Don't want to count it since it could have short-circuited the lookup upon exceeding `expected_max_number_of_operands` operands. It's not really an error either. I think we don't need to worry about pathological cases much (e.g., reads keep returning more and more merge operands) since this is readrandom. But if you or anyone is worried we can make sure it at least doubles in size each time we increase, which should reduce the number of increases much more.

Reply:
My concern is that it can throw all of the statistics off.  For example, if the operation that returns Incomplete is the one that causes all of the blocks to be read and uncompressed, it will take much more time than the following operation that just reads everything from cache.  So your times and percentages will not be accurate.  This might be an outer bounds type case, but those are the ones that are usually very hard to track down.

Reply:
OK, but the operation fails because we didn't know in advance what is the maximum number of operands for a key in the DB. We would only know that value if we scanned the full DB to find out.

Reply:
I think to reflect the real use case more I will retry once for the same key (with resized vector) after getting Status::Incomplete. I'll still count it as one read success because the user usually wants full data, not partial results.

Reply:
My concern was not so much on the success or failure (number of bytes read, etc) but the time of the operation.  You need to figure out how best to account for the time on a retry, and potentially how many retries there are.

Reply:
Added the incomplete result handling strategy to the -help. I think handling incomplete results is a necessary part of using this API so am fine with a "readrandomoperands" operation latency including such handling. The strategy we use here encounters incomplete results extremely infrequently. It is possible the user has a worse or better strategy.